File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/test/ITestResultsObserver.java
package com.laushkina.anastasia.psychosophy.view.test;

import android.content.Context;

import com.laushkina.anastasia.psychosophy.domain.Psychotype;

import java.util.List;

public interface ITestResultsObserver {
    void showExceptionResultDescription();
    void showTypeDescription(Psychotype[] psychotype);

    void showGroupOfQuestions(List<CharSequence> questionSet, boolean isPrevEnabled, int progress, String nextButtonText);
    void showNextButton();

    Context getContext();
}

File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/test/ITestResultsObserver.kt
package com.laushkina.anastasia.psychosophy.view.test

import android.content.Context
import com.laushkina.anastasia.psychosophy.domain.Psychotype

interface ITestResultsObserver {
    fun showExceptionResultDescription()
    fun showTypeDescription(psychotype: Array<Psychotype>)

    fun showGroupOfQuestions(questionSet: List<CharSequence>, isPrevEnabled: Boolean, progress: Int, nextButtonText: String)
    fun showNextButton()

    fun getContext(): Context
}